libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1652,22 +1653,24 @@ string YulUtilFunctions::copyByteArrayToStorageFunction(ArrayType const& _fromTy
 				switch gt(newLen, 31)
 				case 1 {
 					let loopEnd := and(newLen, not(0x1f))
+					<?fromStorage> src := <srcDataLocation>(src) </fromStorage>
```

Comment:
Wow... How this worked for bytes array longer than 31 elements. Probably no such test...